src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,32 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  visibleModal: Todo,
+  setVisibleModal: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  visibleModal, setVisibleModal,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    getUser(visibleModal.userId)
+      .then(setUser)
+      .catch(error => new Error(error.message))
+      .finally(() => setLoading(false));
+  }, [visibleModal.userId]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {loading && !user ? (
```

Comment:
This line can lead to a condition where the Loader is displayed indefinitely. If 'getUser' fails, 'user' will be null and 'loading' will be false. A better check might be just for 'loading' and handle the error state separately